src/dex/uniswap-v3/uniswap-v3-integration.test.ts

```diff
@@ -1239,3 +1239,380 @@ describe('Retro', () => {
+
+      const TokenBSymbol = 'USDC';
+      const TokenB = Tokens[network][TokenBSymbol];
+
+      const amounts = [
+        0n,
+        6000000n,
+        12000000n,
+        18000000n,
+        24000000n,
+        30000000n,
+        36000000n,
+        42000000n,
+        48000000n,
+        54000000n,
+        60000000n,
+        66000000n,
+        72000000n,
+        78000000n,
+        84000000n,
+        90000000n,
+        96000000n,
+        102000000n,
+        108000000n,
+        114000000n,
+        120000000n,
+        126000000n,
+        132000000n,
+        138000000n,
+        144000000n,
+        150000000n,
+        156000000n,
+        162000000n,
+        168000000n,
+        174000000n,
+        180000000n,
+        186000000n,
+        192000000n,
+        198000000n,
+        204000000n,
+        210000000n,
+        216000000n,
+        222000000n,
+        228000000n,
+        234000000n,
+        240000000n,
+        246000000n,
+        252000000n,
+        258000000n,
+        264000000n,
+        270000000n,
+        276000000n,
+        282000000n,
+        288000000n,
+        294000000n,
+        300000000n,
+      ];
+
+      const pools = await baseswapV3.getPoolIdentifiers(
+        TokenA,
+        TokenB,
+        SwapSide.SELL,
+        blockNumber,
+      );
+      console.log(
+        `${TokenASymbol} <> ${TokenBSymbol} Pool Identifiers: `,
+        pools,
+      );
+
+      expect(pools.length).toBeGreaterThan(0);
+
+      const poolPrices = await baseswapV3.getPricesVolume(
+        TokenA,
+        TokenB,
+        amounts,
+        SwapSide.SELL,
+        blockNumber,
+        pools,
+      );
+      console.log(
+        `${TokenASymbol} <> ${TokenBSymbol} Pool Prices: `,
+        poolPrices,
+      );
+
+      expect(poolPrices).not.toBeNull();
+      checkPoolPrices(
+        poolPrices!.filter(pp => pp.unit !== 0n),
+        amounts,
+        SwapSide.SELL,
+        dexKey,
+      );
+
+      // Check if onchain pricing equals to calculated ones
+      let falseChecksCounter = 0;
+      await Promise.all(
+        poolPrices!.map(async price => {
+          const fee = baseswapV3.eventPools[price.poolIdentifier!]!.feeCode;
+          const res = await checkOnChainPricing(
+            dexHelper,
+            baseswapV3,
+            'quoteExactInputSingle',
+            blockNumber,
+            QuoterV2,
+            price.prices,
+            TokenA.address,
+            TokenB.address,
+            fee,
+            amounts,
+          );
+          if (res === false) falseChecksCounter++;
+        }),
+      );
+      expect(falseChecksCounter).toBeLessThan(poolPrices!.length);
+    });
+
+    it('getPoolIdentifiers and getPricesVolume BUY stable pairs', async function () {
+      const TokenASymbol = 'DAI';
+      const TokenA = Tokens[network][TokenASymbol];
+
+      const TokenBSymbol = 'USDC';
+      const TokenB = Tokens[network][TokenBSymbol];
+
+      const amountsBuy = [
```

Comment:
This is duplicated. Maybe it should be a good idea to have one variable.

Reply:
done